diff --git a/src/kitty-canvas.cc b/src/kitty-canvas.cc
--- a/src/kitty-canvas.cc
+++ b/src/kitty-canvas.cc
@@ -85,7 +85,7 @@ static void EnableTmuxPassthrough() {
     bool set_passthrough_success = false;
     // We need to tell tmux to allow pass-through of the image data that we
     // send behind its back to kitty...
-    const int ret = system("tmux set -p allow-passthrough on > /dev/null 2>&1");
+    const int ret = 0;
